### PR TITLE
Dcerpc single tx/v3

### PR DESCRIPTION
**Pull request by inashivb**
DCERPC parser in Rust.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/2779

S-V PR to go with this: https://github.com/OISF/suricata-verify/pull/226

### CODE REVIEW

File: rust/src/common.rs [hunk 1]
```diff
@@ -14,6 +14,24 @@ macro_rules! take_until_and_consume (
...
+  ($item:expr) => {};
+);
+
+
+#[cfg(feature = "debug-validate")]
+#[macro_export]
+macro_rules! debug_validate_bug_on (
+  ($item:expr) => {
+    if $item {
+        panic!("Value should not be NULL!");
```
Comment (victorjulien) [inline rust/src/common.rs:30]: `NULL` isn't right here: it triggers on `false`.

**Comment by victorjulien** [comment]
The commit adding `debug_validate_bug_on` should be in its own commit w/o adding checks to the dcerpc parser. These can be part of the normal parser adding commit.

Can you also expand the git commit messages? They are really sparse right now. Especially the commit adding the Rust dceprc parser itself should explain more about how it works, what it implements and such.

File: rust/src/common.rs (hunk 1 above)
Comment (inashivb) [reply rust/src/common.rs:30]: Do you mean the panic statement should be changed to something else? It gets triggered if header is None.

**Comment by inashivb** [comment]
> The commit adding `debug_validate_bug_on` should be in its own commit w/o adding checks to the dcerpc parser. These can be part of the normal parser adding commit.
> 
> Can you also expand the git commit messages? They are really sparse right now. Especially the commit adding the Rust dceprc parser itself should explain more about how it works, what it implements and such.

Yes sure. Thanks!

File: rust/src/common.rs (hunk 1 above)
Comment (victorjulien) [reply rust/src/common.rs:30]: I think NULL in the string is confusing. It's about a condition being false. NULL suggests a pointer value.

**Comment by inashivb** [comment]
Replaced with #4926

**Pull request closed without merging**